imageio/plugins/rawpy.py

```diff
@@ -0,0 +1,94 @@
+""" Read/Write images using rawpy.
+
+rawpy is an easy-to-use Python wrapper for the LibRaw library. 
+It also contains some extra functionality for finding and repairing hot/dead pixels.
+"""
+
+import rawpy
+import numpy as np
+
+from typing import Any, Callable, Dict, Iterator, List, Optional, Tuple, Union, cast
+from ..core.request import URI_BYTES, InitializationError, IOMode, Request
+from ..core.v3_plugin_api import PluginV3
+from ..typing import ArrayLike
+
+
+class RawPyPlugin(PluginV3):
+    """A class representing the rawpy plugin.
+
+    Methods
+    -------
+
+    .. autosummary::
+    :toctree: _plugins/rawpy
+
+    RawPyPlugin.read
+    """
+    def __init__(self, request: Request) -> None:
+        """Instantiates a new rawpy plugin object
+
+        Parameters
+        ----------
+        request: {Request}
+            A request object representing the resource to be operated on.
+        """
+
+        super().__init__(request)
+
+        self._image_file = None
+
+        if request.mode.io_mode == IOMode.read:
```

Comment:
Since rawpy doesn't support writing, we should check if the plugin is called with `IOMode.write` and if so raise an `InitializationError` with the message that rawpy can't write images.

Reply:
Thank you for your comment. I have updated it now accordingly. I also noticed that `rawpy.imread()` just calls `open_buffer()` without doing any additional thing. In addition to that, I did not use the `with` context manager because then I can not read the file again from the request object using `get_file()` in the read method. Therefore, I have changed my implementation a bit. Please check and let me know what do you think.